### PR TITLE
Yup compact rejector is optional

### PR DESCRIPTION
**Pull request by JulienMalige**
This following code should be accepted.
```
Yup.array()
    .compact()
```
https://github.com/jquense/yup#arraycompactrejector-value--boolean-schema

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/yup/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for yup 0.26
+// Type definitions for yup 0.27
```
Comment (sseppola) [inline types/yup/index.d.ts]: Yup 0.27 has not been released (https://github.com/jquense/yup/releases)
I'm unsure about the versioning of DefinitelyTyped projects, but I don't think you need to bump anything unless the feature you added was specifically for 0.27

**Review by sseppola — COMMENTED** [review]
As far as I can tell the optional rejector was always a part of yup, so if you just reduce the version back to what it was I'm happy to approve this

File: types/yup/index.d.ts (hunk 1 above)
Comment (JulienMalige) [reply types/yup/index.d.ts]: Its a mistake, I was thinking that this number was the version of the current file.

Comment (JulienMalige) [reply types/yup/index.d.ts]: fixed :)

**Review by sseppola — APPROVED** [review]

**Comment by sseppola** [comment]
Thanks for your contribution @JulienMalige :)

**Pull request merged**